### DISCUSSION TITLE
VIA definitions

### DISCUSSION THREAD

**Discussion by gldtn** [vial-kb/vial-gui#203] [Q&A]
I clicked on file > "Download VIA definitions", but nothing happens. When I try to sideload VIA JSON I get a python error
![py_error](https://github.com/vial-kb/vial-gui/assets/981383/f73461fd-e090-4b1e-8ae0-7896ec0bb7b9)

Try to get it to detect my Mode Envoy 

Thanks!

**Comment by dwlocks** [comment]
After poking around in the code, "Download VIA definitions" does in fact download a large json file from the the repo.  I believe the file is at least loaded, but perhaps not processed.
I also tried sideload, but I think the file to be sideloaded is not the VIA definitions.

**Comment by slavkosky** [comment]
I too have run into this same 'vendorId' error while trying to import a VIA JSON keymap/macros, as I switched from VIA fw to Vial. I tried deleting the "vendorProductId" from the VIA JSON but no luck

**Comment by oliarn** [comment]
The last couple of days I also get this message when trying to sideload. 
Has worked flawlessly since January when I started using VIAL instead of VIA.

Same exact message; 

    File "main_window.py", line 343, in on_sideload_json
    File "autorefresh\autorefresh.py", line 61, in sideload_via_json
    File "autorefresh\autorefresh_thread.py", line 92, in sideload_via_json
  
  KeyError: 'vendorId'

Portable version,
Windows 11,
YMDK Air40

**Reply by oliarn** [reply]
Actually got it working by trying repeatedly. May have tried to sideload a faulty Json file previously, but I don't think so.
